info-compiler/src/test/java/com/google/sps/NewsContentExtractorTest.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.webcrawler;
+
+import com.google.sps.data.NewsArticle;
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+import org.apache.tika.exception.TikaException;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.xml.sax.SAXException;
+import static org.mockito.Mockito.*;
+
+/**
+ * A tester for news article content extraction.
+ */
+@RunWith(JUnit4.class)
+public final class NewsContentExtractorTest {
+  private final static String URL =
+    "https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html";
+  private final static String WRONG_URL =
+    "https://www.wrong.com";
+  private final static NewsArticle NEWS_ARTICLE =
+      new NewsArticle(
+          "AOC wins NY Democratic primary against Michelle Caruso-Cabrera, CNN projects - " +
+          "CNNPolitics",
+          URL,
+          "Washington (CNN)Freshman Democratic Rep. Alexandria Ocasio-Cortez will defeat former " +
+          "longtime CNBC correspondent and anchor Michelle Caruso-Cabrera in a Democratic " +
+          "primary election on Tuesday for New York's 14th Congressional District, CNN projects.");
+  private InputStream validWebpageStream;
+
+  @Before
+  public void createInputStream() throws IOException {
+    validWebpageStream = new URL(URL).openStream();
+  }
+
+  @Test
+  public void extractContentFromHtml_validWebpage() {
+    // Extract content and meta data from a valid webpage. The extracted information should be
+    // consistent with {@code URL} and {@code NEWS_ARTICLE}. Content processing hasn't occurred
+    // so the abbreviated content is null.
+    Optional<NewsArticle> potentialNewsArticle =
+        NewsContentExtractor.extractContentFromHtml(validWebpageStream, URL);
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    NewsArticle newsArticle = potentialNewsArticle.get();
```

Comment:
One thing I want to point out: If you had implemented an equals() method in NewsArticle you could just build a "expectedNewsArticle" object and assert that the two objects are equal instead of asserting the validity of each field. This could come in handy the more tests you write (and for future classes as well). It's up to you whether you want to implement this.

Reply:
Got it! I think this will be beneficial for the future and make the code cleaner, since I won't have to explicitly write what fields there are and how many fields there are. I'll make the adjustment.

Reply:
This is done!

Reply:
Looks great, thanks! One nit:you can change the assert to:
Assert.assertEquals(EXPECTED_NEWS_ARTICLE, potentialNewsArticle.get()).
If you look at the Assert Javadoc, the expected argument comes first. I would make that change for the rest of your tests as well.

Reply:
Will do!